src/firmware/guest/mod.rs

```diff
@@ -79,28 +80,29 @@ impl Firmware {
     /// # Arguments
     ///
     /// * `message_version` - (Optional) Used for the SnpGuestRequest, specifies the message version number defaults to 1.
-    /// * `ext_report_request` - an SnpExtReportReq object with its associated data for requesting the extended attestation report.
+    /// * `report_request` - an SnpReportReq object with its associated data.
     ///
     pub fn snp_get_ext_report(
         &mut self,
         message_version: Option<u8>,
-        ext_report_request: SnpExtReportReq,
-    ) -> Result<SnpReportRsp, Indeterminate<Error>> {
+        report_request: SnpReportReq,
+    ) -> Result<(AttestationReport, CertTable), Indeterminate<Error>> {
+        let ext_report_request: SnpExtReportReq = SnpExtReportReq::new(report_request);
```

Comment:
@dubek very good point. Like we talked about during the meeting, I misunderstood the behavior of the IOCTL. I am working on a patch to fix this issue, and will tag you on it when it is ready for review.

Reply:
> Related: We should also consider how we automatically test this. It will include spinning an SNP guest and running the test code in it. Do we have any idea how to approach this?

We could easily use libkrun for running some tests in a CI pipeline. The biggest problem is, as usual, the availability of publicly-accessible SNP-capable systems to run such pipeline.